### PR TITLE
JPA Repository Implemented. Working on exceptions

### PR DESCRIPTION
**Pull request by jobysanthosh**
I need to implement String and null exceptions to be caught.

### CODE REVIEW

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java [hunk 1]
```diff
@@ -1,77 +1,96 @@
...
-			@RequestBody LibraryBranch branch) {
-		boolean check = libBranchService.ifBranchExists(branch.getBranchId());
-		if(check == true && branch!=null) {
-			libBranchService.updateLibraryBranch(branch);
-			return new ResponseEntity<LibraryBranch>(branch, HttpStatus.OK);
-		}
-		else {
-			return new ResponseEntity<LibraryBranch>(branch, HttpStatus.NOT_FOUND);
+	@PutMapping(value = "/{branchId}", consumes = {"application/xml", "application/json"})
+		public ResponseEntity<?> updateLibraryBranch(	@RequestHeader("Accept") String accept,
```
Comment (janfernandosmoothstack) [inline LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java:44]: Good job on the URI's.

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java [hunk 2]
```diff
@@ -1,77 +1,96 @@
...
-	}
 
-	@RequestMapping("/branches/{branchId}/bookId")
-	public List<BookBL> getBooks(@PathVariable(value="branchId") int branchId) {
-		return libBranchService.getBooks(branchId);
-	}
+//		@RequestMapping("/{branchId}/bookId")
+//		public List<Book> getBooks(@PathVariable(value="branchId") int branchId) {
+//			return libBranchService.getBooks(branchId);
+//		}
```
Comment (janfernandosmoothstack) [inline LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java:65]: Remove this code if you do not plan to use it.

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java [hunk 3]
```diff
@@ -1,77 +1,96 @@
...
 
-	@PutMapping(value = "/branches/{branchId}/bookId/{bookId}/copies/{copies}", consumes = {"application/xml", "application/json"})
-	public ResponseEntity<?> updateBookCopy( @RequestHeader("Accept") String accept,
-											@RequestBody BookCopiesBL bookCopy) {
-		
-		boolean branchcheck = libBranchService.ifBranchExists(bookCopy.getBranchId());
-		boolean bookcheck = libBranchService.ifBookExists(bookCopy.getBookId(), bookCopy.getBranchId());
-		
-			if(branchcheck == true  && bookcheck == true && bookCopy!=null) {
-					libBranchService.updateBookCopy(bookCopy);
```
Comment (janfernandosmoothstack) [inline LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java:68]: Your null checking should have OR not AND. You should only go through if all the data is good. If you only check with AND, the process will go through if at least one is okay. But you want it to stop if at least one is null.

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java [hunk 4]
```diff
@@ -1,77 +1,96 @@
...
 	LibraryBranchService libBranchService;
 
-	@GetMapping("/branches")
+	@GetMapping(value="")
 	public List<LibraryBranch> getAllLibraryBranch() {
 		return libBranchService.getAllLibraryBranch();
 	}
 
 
-	@PutMapping(value = "/branches/{branchId}/name/{name}/address/{address}",
```
Comment (sayanastha39) [inline LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java:35]: try changing your path without passing variable in the URI

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java (hunk 2 above)
Comment (sayanastha39) [inline LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java:65]: remove unnecessary comments

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java (hunk 3 above)
Comment (sayanastha39) [reply LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Controller/LibraryBranchController.java:68]: good catch @janfernandosmoothstack Janet, this made me double-check my part of the code.

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/LmsLibrarianApplication.java [hunk 5]
```diff
@@ -1,13 +1,82 @@
...
+import org.apache.tomcat.util.descriptor.web.SecurityConstraint;
 import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory;
+import org.springframework.boot.web.servlet.server.ServletWebServerFactory;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
+//@EnableAutoConfiguration
```
Comment (sayanastha39) [inline LMSLibrarian/src/main/java/com/lms/LMSLibrarian/LmsLibrarianApplication.java:15]: remove if not needed

File: LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Service/LibraryBranchService.java [hunk 6]
```diff
@@ -15,43 +15,50 @@
...
-		return libBranchDao.getBookById(branchId, bookId);
-	}
+//	public List<Book> getBooks(int branchId) {
+//		return libBranchDao.getBooks(branchId);
+//	}
+//	
+//	public Book getBookById(int branchId, int bookId) {
+//		return libBranchDao.getBookById(branchId, bookId);
+//	}
 }
```
Comment (sayanastha39) [inline LMSLibrarian/src/main/java/com/lms/LMSLibrarian/Service/LibraryBranchService.java:64]: remove comments

**Pull request closed without merging**